### PR TITLE
[VERBOSE] XFT_VERBOSE macro support

### PR DESCRIPTION
**Pull request by wenhuanh**
1) XFT_VERBOSE=0 -> no verbose output
2) XFT_VERBOSE=1 -> primitive execution profiling timings 

@wenhuanh 
1）Need to check perf change after XFT_VERBOSE?
2）What is the verbose log?

### CODE REVIEW

File: src/common/my_types.h [hunk 1]
```diff
@@ -235,31 +236,61 @@ class Matrix {
         this->rows = rows;
         this->cols = cols;
         this->stride = m.stride;
+        if (xft_get_verbose() == 2) {
```
Comment (changqi1) [inline src/common/my_types.h:239]: when real allocation

File: src/utils/matmul_helper.h [hunk 2]
```diff
@@ -340,18 +341,51 @@ class MMHelper {
             const float *scaleB, const float *zeroB, float beta, OutT *C, int ldc) {
         // FP32
         if constexpr (std::is_same_v<WeiT, float>) {
-            TimeLine t("xdnn_sgemm_compute");
-            xdnn_sgemm_compute(transA, M, N, K, alpha, A, lda, packedB, beta, C, ldc);
+            if (xft_get_verbose() >= 1) {
```
Comment (changqi1) [inline src/utils/matmul_helper.h:344]: Use Macro to define the logical.

File: src/utils/matmul_helper.h [hunk 3]
```diff
@@ -340,18 +341,51 @@ class MMHelper {
...
+                TimeLine t("xdnn_sgemm_compute");
+                double ms = get_msec();
+                xdnn_sgemm_compute(transA, M, N, K, alpha, A, lda, packedB, beta, C, ldc);
+                ms = get_msec() - ms;
+                printf("xft_verbose,exec,cpu,api,xdnn_sgemm_compute,m%dk%dn%d,%g\n",
```
Comment (changqi1) [inline src/utils/matmul_helper.h:349]: Need to format it

File: src/utils/verbose.h [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+#include <sys/time.h>
+
+static double get_msec() {
+    struct timeval time;
+    gettimeofday(&time, nullptr);
+    return 1e+3 * static_cast<double>(time.tv_sec)
+            + 1e-3 * static_cast<double>(time.tv_usec);
+}
+
+static int xft_get_verbose() {
```
Comment (changqi1) [inline src/utils/verbose.h:17]: need to use class to get verbose value

File: src/utils/verbose.h [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+#define VERBOSE_HPP
+
+#include <cinttypes>
+#include <mutex>
+#include <stdio.h>
+#include <sys/time.h>
+
+static double get_msec() {
+    struct timeval time;
+    gettimeofday(&time, nullptr);
```
Comment (changqi1) [inline src/utils/verbose.h:12]: #include <chrono>

auto tag_0 = std::chrono::high_resolution_clock::now();
decltype(tag_0) tag_1;
static std::chrono::duration<double> diff_0_1 = tag_0 - tag_0;
...
tag_1 = std::chrono::high_resolution_clock::now();
diff_0_1 += tag_1 - tag_0;

Comment (changqi1) [reply src/utils/verbose.h:12]: need to use class

File: src/utils/verbose.h [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ============================================================================
+#pragma once
+#include <cinttypes>
+#include <mutex>
+#include <stdio.h>
```
Comment (changqi1) [inline src/utils/verbose.h:18]: #include <iostream>

File: src/utils/matmul_helper.h [hunk 7]
```diff
@@ -480,6 +480,7 @@ class MMHelper {
     template <typename InT, typename WeiT, typename OutT>
     static void compute(bool transA, int M, int N, int K, float alpha, const InT *A, int lda, const WeiT *packedB,
             const float *scaleB, const float *zeroB, float beta, OutT *C, int ldc) {
+        Env::xft_set_verbose();
```
Comment (changqi1) [inline src/utils/matmul_helper.h:483]: Could call it once at the beginning of the application.

File: src/common/my_types.h [hunk 8]
```diff
@@ -300,6 +304,9 @@ class Matrix {
         this->cols = cols;
         this->stride = stride;
         this->data.Resize(rows, cols, stride);
+
+        Env::xft_set_verbose();
```
Comment (changqi1) [inline src/common/my_types.h:308]: the MatrixVerbose(rows, cols, stride, T) func could in Resize()

File: src/utils/matmul_helper.h (hunk 7 above)
Comment (changqi1) [reply src/utils/matmul_helper.h:483]: https://github.com/intel/xFasterTransformer/blob/main/src/models/models.cpp#L102

Comment (wenhuanh) [reply src/utils/matmul_helper.h]: @changqi1 I revised it in models.h and models.cpp and kept Model::config as its original implement which was @Duyi-Wang 's strong suggestion.

Comment (Duyi-Wang) [reply src/utils/matmul_helper.h]: It should be called in constructor instead of model.config(), which will be called at each generation in python.

Comment (wenhuanh) [reply src/utils/matmul_helper.h]: Thanks Duyi, you also recommend keep two Model::config functions as its original implement.

**Review by changqi1 — APPROVED** [review]

**Comment by wenhuanh** [comment]
1) Performance has validate to be no change on cpx with llama-2-7b
       
2) Log:  XFT_VERBOSE=1 

> xft_verbose,exec,cpu,api,xdnn_sgemm_f32f16f32_compute,m1k4096n22016,5.28777
> xft_verbose,exec,cpu,api,xdnn_sgemm_f32f16f32_compute_residential,m1k11008n4096,2.65826
> xft_verbose,exec,cpu,api,xdnn_sgemm_f32f16f32_compute,m1k4096n12288,2.96771
> xft_verbose,exec,cpu,api,xdnn_sgemm_f32f16f32_compute_residential,m1k4096n4096,0.990329
> xft_verbose,exec,cpu,api,xdnn_sgemm_f32f16f32_compute,m1k4096n22016,5.29279
> xft_verbose,exec,cpu,api,xdnn_sgemm_f32f16f32_compute_residential,m1k11008n4096,2.6402
> xft_verbose,exec,cpu,api,xdnn_sgemm_f32f16f32_compute,m1k4096n12288,2.96864
> xft_verbose,exec,cpu,api,xdnn_sgemm_f32f16f32_compute_residential,m1k4096n4096,0.99039
> xft_verbose,exec,cpu,api,xdnn_sgemm_f32f16f32_compute,m1k4096n22016,5.28667
> xft_verbose,exec,cpu,api,xdnn_sgemm_f32f16f32_compute_residential,m1k11008n4096,2.65359
> xft_verbose,exec,cpu,api,xdnn_sgemm_f32f16f32_compute,m1k4096n12288,2.96422
> xft_verbose,exec,cpu,api,xdnn_sgemm_f32f16f32_compute_residential,m1k4096n4096,0.987523
> xft_verbose,exec,cpu,api,xdnn_sgemm_f32f16f32_compute,m1k4096n22016,5.28914
> xft_verbose,exec,cpu,api,xdnn_sgemm_f32f16f32_compute_residential,m1k11008n4096,2.65679
> xft_verbose,exec,cpu,api,xdnn_sgemm_f32f16f32_compute,m1k4096n12288,2.96964
> xft_verbose,exec,cpu,api,xdnn_sgemm_f32f16f32_compute_residential,m1k4096n4096,0.987794
> xft_verbose,exec,cpu,api,xdnn_sgemm_f32f16f32_compute,m1k4096n22016,5.2912
> xft_verbose,exec,cpu,api,xdnn_sgemm_f32f16f32_compute_residential,m1k11008n4096,2.65284
> xft_verbose,exec,cpu,api,xdnn_sgemm_f32f16f32_compute,m1k4096n12288,2.96652
> xft_verbose,exec,cpu,api,xdnn_sgemm_f32f16f32_compute_residential,m1k4096n4096,0.98681
> xft_verbose,exec,cpu,api,xdnn_sgemm_f32f16f32_compute,m1k4096n22016,4.84878
> xft_verbose,exec,cpu,api,xdnn_sgemm_f32f16f32_compute_residential,m1k11008n4096,2.65204
> xft_verbose,exec,cpu,api,xdnn_sgemm_f32f16f32_compute,m1k4096n32000,2.48132

**Pull request merged**